src/main/Main.hs

```diff
@@ -884,7 +884,7 @@ packagesCmd () go@GlobalOpts{..} =
 targetsCmd :: Text -> GlobalOpts -> IO ()
 targetsCmd target go@GlobalOpts{..} =
     withBuildConfig go $
-    do (_realTargets,_,pkgs) <- ghciSetup Nothing [target]
+    do (_realTargets,_,pkgs) <- ghciSetup (Just BSAll) Nothing [target]
```

Comment:
@chrisdone: Is building the project (and continuing even if it fails) the correct behaviour for `stack targets`?